lib/internal/http2/core.js

```diff
@@ -2038,21 +2043,24 @@ const setTimeout = {
   enumerable: true,
   writable: true,
   value: function(msecs, callback) {
-    if (typeof msecs !== 'number') {
-      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
-                                 'msecs',
-                                 'number');
-    }
+    // Type checking identical to timers.enroll()
+    msecs = validateTimerDuration(msecs);
+
     if (msecs === 0) {
-      unenroll(this);
+      clearTimeout(this[kTimeout]);
+
       if (callback !== undefined) {
         if (typeof callback !== 'function')
           throw new errors.TypeError('ERR_INVALID_CALLBACK');
         this.removeListener('timeout', callback);
       }
     } else {
-      enroll(this, msecs);
-      this[kUpdateTimer]();
+      clearTimeout(this[kTimeout]);
+      this[kTimeout] = setUnrefTimeout(() => {
+        this._onTimeout();
+      }, msecs);
+      if (this[kSession]) this[kSession][kUpdateTimer]();
```

Comment:
Does this actually need the conditional? Couldn't it just call `this[kUpdateTimer]();` like before?

Reply:
that `if (this[kSession])` is roughly equal to `if (this instanceof HTTP2Stream)`

Reply:
It just seems like we could just straight up call `this[kUpdateTimer]()` since it has the same check in it. If this is about avoiding the `_unrefActive` then this call could just be moved before `this[kTimeout]` is declared.

Reply:
What if there is no session then?

There is no point in calling `this[kUpdateTimer]();` on the Stream - you'll just insert the stream timer twice... Might as well move the extra conditional bit here.